src/handlers.py

```diff
@@ -22,12 +26,71 @@
 #   def get(self):
 #     self.render('base.tpl')
 
+
 class CspHandler(handlers.BaseAjaxHandler):
 
-  def post(self):
-    try:
-      report = json.loads(self.request.body)
-      logging.warn('CSP Violation: %s' % (json.dumps(report['csp-report'])))
-      self.render_json({})
-    except:
-      self.render_json({'error': 'invalid CSP report'})
+    def post(self):
+        try:
+            report = json.loads(self.request.body)
+            logging.warn('CSP Violation: %s' % (json.dumps(
+                         report['csp-report'])))
+            self.render_json({})
+        except:
+            self.render_json({'error': 'invalid CSP report'})
+
+
+class MainPage(handlers.BaseHandler):
+    """Serves static index.html with chosen authorization method
+
+    Avaible authorization methods:
+        * basic_auth (default)
+        * google
+
+    Remember:
+        * To choose method edit AUTHORIZATION_METHOD in base/constants.py
+        * [google_method] You need to uncomment login: required in
+          app.yaml.base
+        * [google_method] To update email white list add an email to file:
+          base/constants.py
+    """
+
+    def check_basic_auth(self):
+        return self.request.authorization == constants.BASIC_AUTH
+
+    def check_google_auth(self):
+        user = users.get_current_user()
+        if not user:
+            return False
+        _, auth_domain = user.email().split("@", 1)
+        is_domain = auth_domain in constants.EMAIL_WHITELIST
+        is_email = user.email() in constants.EMAIL_WHITELIST
+        if is_domain or is_email:
+            return True
+
+    def render_with_auth(self, path, _auth_type='basic_auth'):
+        auth_option = {
+            'basic_auth': self.check_basic_auth(),
+            'google': self.check_google_auth()
+        }
+
+        is_authorized = auth_option[_auth_type]
+
+        if is_authorized:
+            return self.render(path)
+
+        if _auth_type == 'basic_auth':
```

Comment:
See comment on line 70, use real polymorphism instead of hardcoding logic into a function parameter